src/sentry/grouptype/grouptype.py

```diff
@@ -0,0 +1,108 @@
+from dataclasses import dataclass
+
+from sentry.event_manager import DEFAULT_GROUPHASH_IGNORE_LIMIT
+from sentry.types.issues import GroupCategory
+
+_group_type_registry = {}
+
+
+@dataclass(frozen=True)
+class GroupType:
+    type_id: int
+    slug: str
+    description: str
+    category: int
+    ignore_limit: int = DEFAULT_GROUPHASH_IGNORE_LIMIT
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        _group_type_registry[cls.__name__] = cls
+
+    def __post_init__(self):
+        valid_categories = [category.value for category in GroupCategory]
+        if self.category not in valid_categories:
+            raise ValueError(f"Category must be one of {valid_categories} from GroupCategory.")
+
+    def get_category_by_slug(self, slug: str):
+        return _group_type_registry[slug].category
+
+
+@dataclass(frozen=True)
+class ErrorGroupType(GroupType):
+    type_id = 1
+    slug = "ERROR"
+    description = "Error"
+    category = GroupCategory.ERROR.value
```

Comment:
think we want `ignore_limit` here at `0`

Reply:
well, we're only using noise reduction for performance issues specifically right now, so not a big deal yet

Reply:
Might as well set it just in case it's missed in the future